### PR TITLE
report(metric-filter): handle multiple reports in the same DOM

### LINKED ISSUE #12725: Audits Tabs are not working in "Desktop" view
### FAQ

- [X] Yes, my issue is not about [variability](https://github.com/GoogleChrome/lighthouse/blob/master/docs/variability.md) or [throttling](https://github.com/GoogleChrome/lighthouse/blob/master/docs/throttling.md).
- [X] Yes, my issue is not about a specific accessibility audit (file with [axe-core](https://github.com/dequelabs/axe-core) instead).

### URL

https://www.youtube.com/ [for Any URL]

### What happened?

image
Screenshot 2021-06-30 at 6 39 18 PM

These audit categories are not clickable in "Desktop" view.

### What did you expect?

Click show work like for "Mobile" view

### What have you tried?

_No response_

### How were you running Lighthouse?

PageSpeed Insights

### Lighthouse Version

8.0.0

### Chrome Version

90.0.4430.97

### Node Version

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by paulirish**
fixes a bug that's live in PSI

found by adam in https://github.com/GoogleChrome/lighthouse/pull/12815#issuecomment-884550283
also by sidverma32. fixes #12725

reviewer: the big change is using a checkbox inside of `<label>` instead of relying on `id` & `for` attributes.

### CODE REVIEW

File: report/renderer/performance-category-renderer.js [hunk 1]
```diff
@@ -312,18 +312,18 @@ export class PerformanceCategoryRenderer extends CategoryRenderer {
       ...filterableMetrics,
     ]);
     for (const metric of filterChoices) {
-      const elemId = `metric-${metric.acronym}`;
-      const radioEl = this.dom.createChildOf(metricFilterEl, 'input', 'lh-metricfilter__radio');
-      radioEl.type = 'radio';
-      radioEl.name = 'metricsfilter';
-      radioEl.id = elemId;
+      const {acronym} = metric;
```
Comment (connorjclark) [inline report/renderer/performance-category-renderer.js:315]: seemed fine without?

File: report/renderer/performance-category-renderer.js [hunk 2]
```diff
@@ -332,9 +332,9 @@ export class PerformanceCategoryRenderer extends CategoryRenderer {
       // Toggle class/hidden state based on filter choice.
       radioEl.addEventListener('input', _ => {
         for (const elem of categoryEl.querySelectorAll('label.lh-metricfilter__label')) {
-          elem.classList.toggle('lh-metricfilter__label--active', elem.htmlFor === elemId);
+          elem.classList.toggle('lh-metricfilter__label--active', elem.dataset.acronym === acronym);
```
Comment (connorjclark) [inline report/renderer/performance-category-renderer.js:335]:
```suggestion
          elem.classList.toggle(
            'lh-metricfilter__label--active', elem.dataset.acronym === acronym);
```

slap a newline here if the reason for `acronym` variable was eslint maxlen

File: report/renderer/performance-category-renderer.js [hunk 3]
```diff
@@ -312,18 +312,18 @@ export class PerformanceCategoryRenderer extends CategoryRenderer {
...
+      const {acronym} = metric;
 
       const labelEl = this.dom.createChildOf(metricFilterEl, 'label', 'lh-metricfilter__label');
-      labelEl.htmlFor = elemId;
       labelEl.title = metric.result && metric.result.title;
-      labelEl.textContent = metric.acronym || metric.id;
+      labelEl.textContent = acronym || metric.id;
+      labelEl.dataset.acronym = acronym;
+
+      const radioEl = this.dom.createChildOf(labelEl, 'input', 'lh-metricfilter__radio');
```
Comment (connorjclark) [inline report/renderer/performance-category-renderer.js:322]: so this is reversing what was done here for tab a11y: https://github.com/GoogleChrome/lighthouse/pull/12552

also I noticed that `radioEl.hidden = 'true'` was a thing before my change and is missing here. is it not needed again?

**Review by connorjclark — COMMENTED** [review]
need https://github.com/GoogleChrome/lighthouse/pull/12815 for full review

File: report/renderer/performance-category-renderer.js [hunk 4]
```diff
@@ -312,18 +312,18 @@ export class PerformanceCategoryRenderer extends CategoryRenderer {
       ...filterableMetrics,
     ]);
     for (const metric of filterChoices) {
-      const elemId = `metric-${metric.acronym}`;
```
Comment (connorjclark) [inline report/renderer/performance-category-renderer.js:315]: another option is to keep the previous id / htmlFor stuff but tack on a unique identifier here. 

```js
const elemId = `metric-${metric.acronym}-${Math.random()}`;
const elemId = `metric-${metric.acronym}-${Math.random()-${Date.now()}`;
```

or somehow wire data from the LHR here.

File: report/renderer/performance-category-renderer.js (hunk 2 above)
Comment (paulirish) [reply report/renderer/performance-category-renderer.js:335]: i know it'd be a very PAUL thing to change code in order to escape eslint maxlen, but i swear that's not why i did that.  😎 
but yeah i'm totes fine with keeping things as `metric.acronym` so back we go. :)

File: report/renderer/performance-category-renderer.js (hunk 3 above)
Comment (paulirish) [reply report/renderer/performance-category-renderer.js:322]: > also I noticed that radioEl.hidden = 'true' was a thing before my change and is missing here.

i didnt add these styles just now but... maybe it was double hidden?

image


i'm nearly certain the hidden affected the tababaility of this, but after trying a few things, i'm mostly thinking going back to id/for is most straightforward

**Comment by paulirish** [comment]
setting back to draft. need #12815 especially since right i'm not seeing tabbing behavior that makes _any_ sense in that testpage... 😖

File: report/renderer/performance-category-renderer.js [hunk 5]
```diff
@@ -311,8 +311,9 @@ export class PerformanceCategoryRenderer extends CategoryRenderer {
       ({acronym: 'All'}),
       ...filterableMetrics,
     ]);
+    const nowTs = Date.now();
```
Comment (connorjclark) [inline report/renderer/performance-category-renderer.js:314]: This defiantly deserves a comment about how radio inputs need unique ids and that this is needed to support multiple reports on the same page :)

Comment (connorjclark) [reply report/renderer/performance-category-renderer.js:314]: > defiantly

😠 defiance!

File: report/renderer/performance-category-renderer.js [hunk 6]
```diff
@@ -311,8 +311,9 @@ export class PerformanceCategoryRenderer extends CategoryRenderer {
...
     for (const metric of filterChoices) {
-      const elemId = `metric-${metric.acronym}`;
+      const elemId = `metric-${metric.acronym}-${nowTs}`;
```
Comment (brendankenny) [inline report/renderer/performance-category-renderer.js:319]: FWIW we have a utility for this over in [`Util.getUniqueSuffix`](https://github.com/GoogleChrome/lighthouse/blob/d3504385b8c91a89e6cb6fe590c075dba92adb0c/report/renderer/util.js#L518-L526), used for unique IDs for SVG references in pwa-category-renderer: https://github.com/GoogleChrome/lighthouse/blob/d3504385b8c91a89e6cb6fe590c075dba92adb0c/report/renderer/pwa-category-renderer.js#L167-L172

**Comment by paulirish** [comment]
> need #12815 for full review

In the meantime, here's the combo: https://famous-horses.surge.sh/%E2%8C%A3.psi.english/index.html

**Review by connorjclark — APPROVED** [review]

**Pull request merged**